src/config.rs

```diff
@@ -0,0 +1,383 @@
+    }
+
+    fn get_target_var(&self, target: &Target, key: &str) -> Option<String> {
+        self.get_var(&self.build_var_name(&Environment::target_path(target, key)))
+    }
+
+    fn xargo(&self, target: &Target) -> Result<(Option<bool>, Option<bool>)> {
+        let (build_xargo, target_xargo) = (
+            self.get_build_var("XARGO"),
+            self.get_target_var(target, "XARGO"),
+        );
+        let build_env = if let Some(build) = build_xargo {
+            Some(build.parse::<bool>().or_else(|_| {
+                Err(format!(
+                    "error parsing {} from XARGO environment variable",
+                    build
+                ))
+            })?)
+        } else {
+            None
+        };
+        let target_env = if let Some(t) = target_xargo {
+            Some(t.parse::<bool>().or_else(|_| {
+                Err(format!(
+                    "error parsing {} from XARGO environment variable",
+                    t
+                ))
+            })?)
+        } else {
+            None
+        };
+
+        Ok((build_env, target_env))
+    }
+    fn image(&self, target: &Target) -> Option<String> {
+        self.get_target_var(target, "IMAGE")
+    }
+    fn runner(&self, target: &Target) -> Option<String> {
+        self.get_target_var(target, "RUNNER")
+    }
+
+    fn passthrough(&self, target: &Target) -> (Option<Vec<String>>, Option<Vec<String>>) {
+        let target_envs = self.get_target_var(target, "env_passthrough");
+        let mut passthrough_target = None;
+        if let Some(envs) = target_envs {
+            passthrough_target = Some(vec![]);
+            passthrough_target
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+        let build_envs = self.get_build_var("env_passthrough");
+        let mut passthrough_build = None;
+        if let Some(envs) = build_envs {
+            passthrough_build = Some(vec![]);
+            passthrough_build
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+
+        (passthrough_build, passthrough_target)
+    }
+
+    fn volumes(&self, target: &Target) -> (Option<Vec<String>>, Option<Vec<String>>) {
+        let target_envs = self.get_target_var(target, "ENV_VOLUMES");
+        let mut volumes_target = None;
+        if let Some(envs) = target_envs {
+            volumes_target = Some(vec![]);
+            volumes_target
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
+        }
+        let build_envs = self.get_build_var("ENV_VOLUMES");
+        let mut volumes_build = None;
+        if let Some(envs) = build_envs {
+            volumes_build = Some(vec![]);
+            volumes_build
+                .as_mut()
+                .map(|vec| vec.extend(envs.split_whitespace().map(|v| v.to_string())));
```

Comment:
`split_whitespace` will break volume paths containing spaces. I'd suggest using `:` as separator, same as e.g. `PATH`.

Reply:
In my understanding toml and env should provide the same API to the user. The volume values only describe the environment variable names under which the path that is bound to docker is found. This would be the same regardless if config is done with toml or env. 
This lookup is done in docker.rs:
https://github.com/rust-embedded/cross/blob/4f213ba07b214d36225be63e5b72a6346937a699/src/docker.rs#L133-L137
(changes then from toml to config)
https://github.com/olitha/cross/blob/423dea64dbf046f642274173138d392c91a0d16f/src/docker.rs#L131-L136
So I think volumes can be split by whitespaces as well.

Reply:
Yes, you're right.